### PR TITLE
Document missing TypeInterface feature

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Add small explanation and example.
That way it should be possible to make use of this feature in custom
extensions.

Releases: master, 10.4
Relates: #393

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
Valid since many versions, including 8.7, 9.5, 10.4.

File: Documentation/b-ExtbaseReference/Index.rst [hunk 1]
```diff
@@ -713,6 +713,39 @@ All classes of the domain model must inherit from one of the following two class
...
+
+       use TYPO3\CMS\Core\Type\TypeInterface;
+
+       class ModelName implements TypeInterface
+       {
+           private string $serialized = '';
+
+           public function __construct(string $serialized)
+           {
+               $this->serialized = $serialized;
```
Comment (mbrodala) [inline Documentation/b-ExtbaseReference/Index.rst:735]: We should unserialize the string here (preferrably using `json_decode()`) and use `array` as private value.

File: Documentation/b-ExtbaseReference/Index.rst [hunk 2]
```diff
@@ -713,6 +713,39 @@ All classes of the domain model must inherit from one of the following two class
...
+           }
+
+           // Add proper getter, e.g. for template …
+
+           public function __toString(): string
+           {
+               return $this->serialized;
```
Comment (mbrodala) [inline Documentation/b-ExtbaseReference/Index.rst:742]: Here we then need to serialize again of course.

File: Documentation/b-ExtbaseReference/Index.rst [hunk 3]
```diff
@@ -713,6 +713,39 @@ All classes of the domain model must inherit from one of the following two class
...
+       class ModelName implements TypeInterface
+       {
+           private string $serialized = '';
+
+           public function __construct(string $serialized)
+           {
+               $this->serialized = $serialized;
+           }
+
+           // Add proper getter, e.g. for template …
```
Comment (mbrodala) [inline Documentation/b-ExtbaseReference/Index.rst:738]: We should add a more descriptive example here.

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Comment by linawolf** [comment]
@mbrodala @DanielSiepmann  What is the status of this PR? Can you resolve the open comments?

**Comment by linawolf** [comment]
@mbrodala @DanielSiepmann please get this PR resolved or closed.

**Comment by linawolf** [comment]
@DanielSiepmann , @mbrodala I put this in draft. Please decide how to go on with this PR

**Review by mbrodala — APPROVED** [review]

**Comment by mbrodala** [comment]
I've integrated JSON encode/decode now for a more real-world example.

**Pull request merged**